### PR TITLE
Added support for rivers, along with a map that uses them.

### PR DESCRIPTION
**Pull request by ThatIsAUsername**

### CODE REVIEW

**Comment by ThatIsAUsername** [comment]
Fixes #101

File: src/Terrain/TerrainType.java [hunk 1]
```diff
@@ -96,6 +96,11 @@ private TerrainType(int attributeFlags, int defense, Color mainColor)
   private static final Color REEF_COLOR = new Color(218, 152, 112);
   public static final TerrainType REEF = new TerrainType( REEF_FLAGS, REEF_DEFENSE, REEF_COLOR );
 
+  private static final int RIVER_FLAGS = LAND; // It's not really a WATER type, since boats can't go here.
```
Comment (ThislsAUsername) [inline src/Terrain/TerrainType.java]: It might be amusing to change that, but fair enough.

**Comment by ThislsAUsername** [comment]
I like the new factory, it's much less city-y

**Comment by ThatIsAUsername** [comment]
Hah, I was wondering if you would notice the new factory. Pretty sure you missed it when I swapped out the HQ a little while back...

**Comment by ThislsAUsername** [comment]
It shows up in the diffs. How would I miss it?
I just prefer to avoid criticizing art, since there's so much preference tied up in it. (though I guess code style is much the same)

**Pull request merged**